src/package.json

```diff
@@ -39,6 +39,10 @@
     },
     "./plugin": {
       "require": "./dist/plugin/index.js"
+    },
+    "./route.shared": {
```

Comment:
We can use a new file that controls and makes explicit which epxorts are public.

```ts
// src/route.ts

export { RouteMatch, PageRoute, setCustomRouter, /*...*/ } from './route.shared'
```